lib/rate-limit.js

```diff
@@ -1,3 +1,5 @@
+let attempt = 0
```

Comment:
Does ist make sense to have the attempt counter for all requests together? Shouldn't it be per request?